### PR TITLE
Global usage of authenticated.guard

### PR DESCRIPTION
**Pull request by eprei**
The 'authenticated.guard' guard is now used globally throughout the backend. If you want it not to be used on a certain endpoint you must use the @Public decorator.

Docs:
https://docs.nestjs.com/security/authentication#enable-authentication-globally

### CODE REVIEW

**Review by RobinBurri — APPROVED** [review]
Nickel!

**Review by t-h2o — APPROVED** [review]
:shield:

File: backend/src/auth/auth.controller.ts [hunk 1]
```diff
@@ -38,29 +40,33 @@ export class AuthController {
...
 
+    @Public()
     @Get('loginStatus')
     getStatus(@Req() req) {
         return this.userService.logStatus(req)
     }
 
+    @Public()
     @Post('2fa/turn-on')
     @UseGuards(Activate2faGuard)
```
Comment (t-h2o) [inline backend/src/auth/auth.controller.ts:51]: Why there is `@Public` and `@UseGuards` ?

Comment (eprei) [reply backend/src/auth/auth.controller.ts:51]: @Public is for not using the global guard 'authenticated.guard'. 

@UseGuards(Activate2faGuard) is to use a different guard called Activate2faGuard which verifies that the user is able to activate the 2fa. Namely, the user must be logged in and must not have already activated the 2fa.

Comment (t-h2o) [reply backend/src/auth/auth.controller.ts:51]: Thanks

**Review by shaolin-peanut — APPROVED** [review]
let's goo

**Pull request merged**